packages/starknet-snap/src/index.ts

```diff
@@ -153,6 +153,21 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
       saveMutex,
     };
 
+    // Define mappings to ensure backward compatibility with previous versions of the API.
+    // These mappings replace deprecated parameter names with the updated equivalents,
+    // allowing older integrations to function without changes, while the new version
+    // expects parameters such as `calls`, `details`, and `address`.
+    const paramMappings: Record<string, string> = {
+      signerAddress: 'address',
+      senderAddress: 'address',
+      txnInvocation: 'calls',
+      invocationsDetails: 'details',
+      transaction: 'details',
+    };
+
+    // Apply the mappings to apiParams.requestParams
+    mapDeprecatedParams(apiParams.requestParams, paramMappings);
```

Comment:
the mapping is override every api, as some api is using the same params, as the new API is using OOP, why not implement in those API?